public/app/plugins/datasource/grafana-azure-monitor-datasource/utils/logging.ts

```diff
@@ -0,0 +1,32 @@
+import { AzureMonitorQuery, AzureQueryType } from '../types';
+
+export const getAzureMonitorEvent = (target: AzureMonitorQuery) => {
+  const commonProps = {
+    query_type: target.queryType,
+  };
+  switch (target.queryType) {
+    case AzureQueryType.AzureMonitor:
+      return {
+        ...commonProps,
+        dimensions: target.azureMonitor?.dimensionFilters?.length ?? 0,
+        alias: !!target.azureMonitor?.alias,
+        top: !!target.azureMonitor?.top,
+      };
+      break;
+    case AzureQueryType.LogAnalytics:
+      return {
+        ...commonProps,
+        complexity: target.azureLogAnalytics?.query?.match(/\|/g)?.length,
+        format: target.azureLogAnalytics?.resultFormat,
+      };
+      break;
+
+    case AzureQueryType.AzureResourceGraph:
+      return {
+        ...commonProps,
+        complexity: target.azureResourceGraph?.query?.match(/\|/g)?.length,
+      };
+      break;
+  }
+  return commonProps;
```

Comment:
nit: the `break`s in the switch aren't needed with the returns and you could have `default: return commonProps` at the end